core/drand_test.go

```diff
@@ -427,280 +568,261 @@ func TestDrandPublicStream(t *testing.T) {
+
+	err := dt.WaitUntilRound(t, dt.nodes[0],1)
+	require.NoError(t, err)
+
 	// do a few periods
 	for i := 0; i < 3; i++ {
-		dt.MoveTime(group.Period)
+		dt.AdvanceMockClock(t, group.Period)
+
+		err = dt.WaitUntilRound(t,dt.nodes[0], uint64(i+2))
+		require.NoError(t, err)
 	}
 
 	cm := root.drand.opts.certmanager
 	client := net.NewGrpcClientFromCertManager(cm)
+
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
+
 	// get last round first
+	t.Log("Getting the last round first with PublicRand method")
 	resp, err := client.PublicRand(ctx, rootID, new(drand.PublicRandRequest))
 	require.NoError(t, err)
 
 	//  run streaming and expect responses
 	req := &drand.PublicRandRequest{Round: resp.GetRound()}
 	respCh, err := client.PublicRandStream(ctx, root.drand.priv.Public, req)
 	require.NoError(t, err)
+
 	// expect first round now since node already has it
+	t.Log("Waiting to receive the first round as the node should have it now...")
 	select {
 	case beacon := <-respCh:
+		t.Logf("First round rcv %d \n", resp.GetRound())
 		require.Equal(t, beacon.GetRound(), resp.GetRound())
+
 	case <-time.After(100 * time.Millisecond):
-		require.True(t, false, "too late")
+		t.Logf("First round NOT rcv. Timeout has passed \n")
+		require.True(t, false, "too late for the first round, it didn't reply in time")
 	}
+
 	nTry := 4
 	// we expect the next one now
 	initRound := resp.Round + 1
 	maxRound := initRound + uint64(nTry)
-	fmt.Println("Streaming for future rounds starting from", initRound)
+	t.Logf("Streaming for future rounds starting from %d", initRound)
+
 	for round := initRound; round < maxRound; round++ {
 		// move time to next period
-		dt.MoveTime(group.Period)
+		dt.AdvanceMockClock(t, group.Period)
+
 		select {
 		case beacon := <-respCh:
 			require.Equal(t, beacon.GetRound(), round)
 		case <-time.After(1 * time.Second):
-			require.True(t, false, "too late for streaming, round %d didn't reply in time", round)
+			t.Logf("Round %d NOT rcv. Timeout has passed \n", round)
+			require.True(t, false, fmt.Sprintf("too late for streaming, round %d didn't reply in time", round))
 		}
 	}
+
 	// try fetching with round 0 -> get latest
+	t.Logf("Streaming for rounds starting from %d to %d", 0, maxRound)
+
 	respCh, err = client.PublicRandStream(ctx, root.drand.priv.Public, new(drand.PublicRandRequest))
 	require.NoError(t, err)
+
 	select {
 	case <-respCh:
-		require.False(t, true, "shouldn't get an entry if time doesn't go by")
+		require.False(t, true, "shouldn't get a round if time doesn't go by")
+
 	case <-time.After(50 * time.Millisecond):
 		// correct
 	}
 
-	dt.MoveTime(group.Period)
+	dt.AdvanceMockClock(t, group.Period)
 	select {
 	case resp := <-respCh:
+		t.Logf("Round %d rcv \n", maxRound)
 		require.Equal(t, maxRound, resp.GetRound())
+
 	case <-time.After(50 * time.Millisecond):
 		// correct
 	}
 }
-func TestDrandFollowChain(tt *testing.T) {
+
+// ??????????
+func TestDrandFollowChain(t *testing.T) {
```

Comment:
This test makes sure the "FollowChain" grpc method works fine

Reply:
Great!